src/3rd_party_repos.c

```diff
@@ -574,4 +574,28 @@ enum swupd_code third_party_process_binaries(struct list *files, const char *msg
 	return ret_code;
 }
 
+enum swupd_code third_party_remove_binary(char *filename)
+{
+	enum swupd_code ret_code = SWUPD_OK;
+	int ret;
+	char *script = NULL;
+	char *binary = NULL;
+
+	script = third_party_get_binary_path(sys_basename(filename));
+	binary = sys_path_join(globals.path_prefix, filename);
+
+	if (sys_file_exists(script) && !sys_file_exists(binary)) {
```

Comment:
To avoid a TOCTOU error don't check if the script exists. If the binary doesn't exist just remove it.

Reply:
good idea, done